### DISCUSSION TITLE
multi level array data (embedded in embedded ..)

### DISCUSSION THREAD

**Discussion by SPSpisoft** [isar/isar#1287] [Help]
I am going to store a series of data that I receive in layers and in Jason array format in the database.

Similar to this :

**@collection**
class Email {
  Id id = Isar.autoIncrement; 
  String? title;
 List<<Recipient>Recipient>? recipients;
}

@embedded
class Recipient {
  String? name;
  String? address;
  List<<PhoneNumber>PhoneNumber>? Phones;
}

> 

> 

@embedded
class PhoneNumber{
  String? title;
  String? number;
}

I considered the main class (**Email**)  **_@collection_** and its subclass (**Recipient**) _@embedded_ . So far, the data is stored correctly.. but for the next subclasses (**PhoneNumber**), the null value is stored
.Please help me to solve the problem

**Comment by JonathanSchndr** [comment]
I made the same mistake because I thought that it was slimmer with Embedded than with Link in order not to have a new type. Unfortunately, I couldn't solve the problem either.

**Comment by Scerbelo12** [comment]
Any update on this one? I'm facing the same problem. In the specs it's written that there is no limit to how deep in the tree a nested object can be
